include/multipass/platform.h

```diff
@@ -36,13 +36,22 @@ namespace multipass
 {
 namespace platform
 {
+std::map<QString, QString> extra_settings_defaults();
```

Comment:
That was kind of what I was going for here. This avoids having to mix extra inheritance levels into the singleton. It also allows further settings that depend on conditions other than the platform (for instance, we could envision virtualbox-only settings or something). 

Also, I think extension is more suitable as only common things need be shared. That way, when you read the code, you know they apply everywhere.

Reply:
The problem is that I can easily see the need for overriding defaults per platform, not just adding them. Not sure yet what would have precedence - platform or driver, or if we should define it (but can we not?)… It's not a big deal as it's our internal API, but still :)

Reply:
The way I see it, any defaults that needed to be overridden by platform should not be in the global defaults in the first place. Just place them in the platforms and it's done, right?

Reply:
The problem there is that if just one platform / driver needs to override something, suddenly all the others need to keep the default that could otherwise be global.

Reply:
OK, that is a downside if such situations arise. But we could also do it differently when needed, wouldn't that be preferable? You'd keep the advantage of having globals really applying globally until then, which I think is cleaner. Otherwise, what exactly are you proposing?

Reply:
I would make the platform defaults override the global ones rather tha only add to it.

Reply:
Ah ok, so you just want something like python map's update instead of the [insert here](https://github.com/canonical/multipass/blob/15d1e100fe6e9006c303618336f66ebcc714fbcc/src/utils/settings.cpp#L52), correct?

Reply:
Yes.